dl/src/main/scala/com/intel/analytics/bigdl/optim/Optimizer.scala

```diff
@@ -49,13 +49,6 @@ abstract class Optimizer[T: ClassTag, D](
   protected var comupteThresholdbatchSize: Int = 100
   protected var warmupIterationNum: Int = 200
 
-  def disableCheckSingleton(): this.type = {
-    this.checkSingleton = false
-    this
-  }
-
-  protected var checkSingleton = true
-
```

Comment:
Instead of removing `disableCheckSingleton`, maybe we should mark it as deprecated, and report a warning to ask users to use the "bigdl.check.singleton" config instead.